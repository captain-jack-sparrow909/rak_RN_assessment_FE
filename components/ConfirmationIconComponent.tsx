import { View, Image, StyleSheet } from 'react-native';


const ConfirmationIconComponent = () => {
  return (
    <View>
      <Image
        source={require('../assets/success.png')}
        style={styles.icon}
      />
    </View>
  );
};

const styles = StyleSheet.create({
    icon: {
        width: 220,
        height: 220,
      },
});

export default ConfirmationIconComponent;
