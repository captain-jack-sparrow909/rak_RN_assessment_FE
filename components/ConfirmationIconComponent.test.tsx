import React from 'react';
import { render } from '@testing-library/react-native';
import ConfirmationIconComponent from './ConfirmationIconComponent';

describe('ConfirmationIconComponent', () => {
  it('should match the snapshot', () => {
    const { toJSON } = render(<ConfirmationIconComponent />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render the Image component with correct source and style', () => {
    const { getByTestId } = render(<ConfirmationIconComponent />);
    const image = getByTestId('confirmation-icon');
    expect(image.props.source).toEqual(require('../assets/success.png'));
    expect(image.props.style).toEqual({ width: 220, height: 220 });
  });
});