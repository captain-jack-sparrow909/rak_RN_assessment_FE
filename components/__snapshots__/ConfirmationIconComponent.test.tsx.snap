// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationIconComponent should match the snapshot 1`] = `
<View>
  <Image
    source={1}
    style={
      {
        "height": 220,
        "width": 220,
      }
    }
    testID="confirmation-icon"
  />
</View>
`;
